server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Mailbox.scala

```diff
@@ -118,10 +118,10 @@ object MailboxName {
   type MailboxNameConstraint = NonEmpty
   type MailboxName = String Refined MailboxNameConstraint
 
-  def liftOrThrow(value: String): MailboxName =
+  def validate(value: String): Either[String, MailboxName] =
     refined.refineV[MailboxNameConstraint](value) match {
-      case scala.util.Right(value) => value
-      case Left(error) => throw new IllegalArgumentException(error)
+      case Left(error) => Left(error)
```

Comment:
you do nothing in the match anymore.

Also, we had cases where having `liftOrThrow` made sense when doing the mob

Reply:
That was for the `UnsignedInt` (and I left that case there). For the mailboxName, the only place where it's being refined is in `MailboxFactory`

But yes I could just return in fact  `refined.refineV[MailboxNameConstraint](value)`, no need of a match